AstarteDeviceSDKCSharp/Data/AstarteFailedMessageEntry.cs

```diff
@@ -54,7 +54,7 @@ public AstarteFailedMessageEntry(int qos, byte[] payload, string topic)
             Qos = qos;
```

Comment:
If `relativeExpiry` is `<= 0` means the persistent cache never expires, so you have to cover also this use case.